plugins/inputs/jboss/jboss.go

```diff
@@ -0,0 +1,1222 @@
+				errorChannel <- errors.New("Error decoding JSON response")
+			}
+			// This struct apply on EAR files
+			for typeName, value := range deploy.Result.Subdeployment {
+				if value == nil {
+					log.Printf("D! JBoss Deployment WARNING Subdeployment value is NULL")
+					continue
+				}
+
+				t := value.(map[string]interface{})
+				if t["subsystem"] == nil {
+					log.Printf("D! JBoss Deployment WARNING SUBDEPLOYMENT Subsystem is NULL")
+					continue
+				}
+				subsystem := t["subsystem"].(map[string]interface{})
+
+				if ejbValue, ok := subsystem["ejb3"]; ok {
+					ejb := ejbValue.(map[string]interface{})
+					tags := map[string]string{
+						"jboss_host":   host,
+						"jboss_server": serverName,
+						"name":         typeName,
+						"runtime_name": deploy.Result.RuntimeName,
+					}
+					h.processEJBAppStats(acc, ejb, tags)
+				}
+
+				if webValue, ok := subsystem["web"]; ok {
+					web := webValue.(map[string]interface{})
+					tags := map[string]string{
+						"jboss_host":   host,
+						"jboss_server": serverName,
+						"name":         typeName,
+						"runtime_name": deploy.Result.RuntimeName,
+					}
+					h.processWebAppStats(acc, web, tags)
+				}
+			}
+			// This struct apply on WAR files
+			for typeName, value := range deploy.Result.Subsystem {
+				if value == nil {
+					log.Printf("D! JBoss Deployment SUBSYSTEM  value NULL")
+					continue
+				}
+				if typeName == "web" {
+					web := value.(map[string]interface{})
+					tags := map[string]string{
+						"jboss_host":   host,
+						"jboss_server": serverName,
+						"name":         deploy.Result.Name,
+						"runtime_name": deploy.Result.RuntimeName,
+					}
+					h.processWebAppStats(acc, web, tags)
+				} else {
+					log.Printf("W! JBoss Deployment WAR  from type %s", typeName)
+				}
+			}
+
+		}(deployment)
+	}
+
+	wg.Wait()
+	close(errorChannel)
+
+	// Get all errors and return them as one giant error
+	errorStrings := []string{}
+	for err := range errorChannel {
+		errorStrings = append(errorStrings, err.Error())
+	}
+
+	if len(errorStrings) == 0 {
+		return nil
+	}
+	return errors.New(strings.Join(errorStrings, "\n"))
+}
+
+// Flatten JSON hierarchy to produce field name and field value
+// Parameters:
+//    item: Item map to flatten
+//    fields: Map to store generated fields.
+//    id: Prefix for top level metric (empty string "")
+// Returns:
+//    void
+func (h *JBoss) flatten(item map[string]interface{}, fields map[string]interface{}, id string) {
+	if id != "" {
+		id = id + "_"
+	}
+	for k, i := range item {
+		switch i.(type) {
+		case int:
+			fields[id+k] = i.(float64)
+		case float64:
+			fields[id+k] = i.(float64)
+		case map[string]interface{}:
+			h.flatten(i.(map[string]interface{}), fields, id+k)
+		default:
+		}
+	}
+}
+
+//func (j *JBoss) prepareRequest(optype int, adress map[string]interface{}) (map[string]interface{}, error) {
+func (h *JBoss) prepareRequest(optype int, adress OrderedMap) (map[string]interface{}, error) {
```

Comment:
Function name seem wrong, should be something more like: `createRequestBody`

typo adress -> address

Reply:
ok!